packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -104,7 +104,7 @@ module.exports = function watchmanCrawl(
 
             clocks[root] = response.clock;
             response.files.forEach(fileData => {
-              const name = root + path.sep + fileData.name;
+              const name = path.normalize(path.join(root, fileData.name));
```

Comment:
Do you mind doing some benchmarking on what is fastest on node8? Things may have changed and the path module was rewritten for node 6 also.